### DISCUSSION TITLE
lottie-spm xcframework for Xcode 16 Beta 1

### DISCUSSION THREAD

**Discussion by freak4pc** [airbnb/lottie-ios#2433] [General]
Unfortunately lottie-spm doesn't have an "Issues" tab so I opened it here, hope that's OK :) 

I know Xcode 16 is still in beta, but I'm wondering if it'd be possible / worthwhile to release "Pre-release" versions on the lottie-spm repo that have xcframework(s) for the beta's toolchain. Otherwise, the only choice is to drop back to the source code with lottie-ios :) 

Thank you 🙏

**Comment by calda** [comment]
What is the issue preventing lottie-spm from working with Xcode 16? My expectation is that it should be forwards-compatible with new Xcode versions.

**Comment by freak4pc** [comment]
> What is the issue preventing lottie-spm from working with Xcode 16? My expectation is that it should be forwards-compatible with new Xcode versions.

I can try again and show the error message, but I believe it's because it's a different toolchain and Xcode version, like you specified here:
https://github.com/airbnb/lottie-spm/blob/main/Package.swift#L5-L12

**Comment by calda** [comment]
These XCFrameworks are compiled in a way that is supposed to make them forwards-compatible with any version of Xcode / Swift equal to or more recent than the version that compiled the XCFramework (they have [`BUILD_LIBRARY_FOR_DISTRIBUTION=YES` enabled](https://github.com/airbnb/lottie-ios/blob/master/Rakefile#L71)). I expect an XCFramework built by Xcode 15.2 with `BUILD_LIBRARY_FOR_DISTRIBUTION=YES` to be supported by Xcode 16.

**Comment by calda** [comment]
There was a similar issue / discussion here: https://github.com/airbnb/lottie-ios/issues/2424

It seems like doing a clean build or clearing derived data fixed the issue there.

**Reply by freak4pc** [reply]
Would be nice if I could get to reply before this being locked, but this doesn't seem to be the same case :) 
I'm 100% on a clean build.

<img width="1496" alt="image" src="https://github.com/airbnb/lottie-ios/assets/605076/140847e3-bbbe-4b4c-a07f-9f15e98cf4f2">

**Reply by freak4pc** [reply]
(I tried nuking my dervied data multiple times, and switching to lottie-ios does work)

**Reply by freak4pc** [reply]
The interesting question is why it thinks it was built on Swift 5.6 when you say it's Swift 5.9 🤔

**Reply by freak4pc** [reply]
The swiftinterface file looks right, the only guess I have is I'm getting the wrong binary, somehow. I'm taking another look ...
<img width="842" alt="image" src="https://github.com/airbnb/lottie-ios/assets/605076/2e239789-e78b-4b7a-af78-cc50b0d17ac0">

**Reply by freak4pc** [reply]
Was linked to 4.3.3 instead of 4.4.3 🤦 Classic PEBKAC. My apologies.
